apps/emqx_authn/src/simple_authn/emqx_authn_redis.erl

```diff
@@ -96,51 +96,29 @@ refs() ->
 create(_AuthenticatorID, Config) ->
     create(Config).
 
-create(
-    #{
-        cmd := Cmd,
-        password_hash_algorithm := Algorithm
-    } = Config
-) ->
-    ok = emqx_authn_password_hashing:init(Algorithm),
-    try
-        NCmd = parse_cmd(Cmd),
-        ok = emqx_authn_utils:ensure_apps_started(Algorithm),
-        State = maps:with(
-            [password_hash_algorithm, salt_position],
-            Config
-        ),
-        ResourceId = emqx_authn_utils:make_resource_id(?MODULE),
-        NState = State#{
-            cmd => NCmd,
-            resource_id => ResourceId
-        },
-        {ok, _Data} = emqx_resource:create_local(
-            ResourceId,
-            ?RESOURCE_GROUP,
-            emqx_connector_redis,
-            Config,
-            #{}
-        ),
-        {ok, NState}
-    catch
-        error:{unsupported_cmd, _Cmd} ->
-            {error, {unsupported_cmd, Cmd}};
-        error:missing_password_hash ->
-            {error, missing_password_hash};
-        error:{unsupported_fields, Fields} ->
-            {error, {unsupported_fields, Fields}}
-    end.
-
-update(Config, State) ->
-    case create(Config) of
-        {ok, NewState} ->
-            ok = destroy(State),
-            {ok, NewState};
+create(Config0) ->
+    ResourceId = emqx_authn_utils:make_resource_id(?MODULE),
+    {Config, State} = parse_config(Config0),
```

Comment:
It looks like in the previous code that an `{error, Reason}` tuple is returned if anything goes wrong during the parsing. The error code is still in place within the `parse_config` method but it isn't checked at this point and so if there is an error returned we're gonna just go ahead and keep trying to create the resource without the expected config.

Reply:
Thanks.